tests/app/one_click_unsubscribe/test_one_click_unsubscribe.py

```diff
@@ -119,3 +136,39 @@ def test_invalid_one_click_unsubscribe_url_notification_id(client, sample_email_
     response_json_data = response.get_json()
     assert response.status_code == 404
     assert response_json_data["message"] == {"unsubscribe request": "This is not a valid unsubscribe link."}
+
+
+def test_is_duplicate_unsubscribe_request_for_non_duplicate_request_1(sample_service):
+    # Test case is when the notification_id does not exist in the unsubscribe_request table
+    result = is_duplicate_unsubscribe_request("9d328a7a-d3f4-4494-a429-63525e7338f4")
+    assert result is False
+
+
+def test_is_duplicate_unsubscribe_request_for_non_duplicate_request_2(sample_service):
+    # Test case is an unsubscribe request that has the same notification_id as a previous request
+    # that has been processed by the service. Only sequential unprocessed unsubscribe requests with the same
+    # notification_id are being considered as duplicate requests.
+    notification_id = create_unsubscribe_request_and_return_the_notification_id(
+        sample_service, is_batched=True, processed_by_service=True
+    )
+    # Simulate a duplicate unsubscribe request with the same notification_id
+    result = is_duplicate_unsubscribe_request(notification_id)
+    assert result is False
+
+
+def test_is_duplicate_unsubscribe_request_for_un_batched_request(sample_service):
+    notification_id = create_unsubscribe_request_and_return_the_notification_id(
+        sample_service, is_batched=False, processed_by_service=False
+    )
+    # Simulate a duplicate unsubscribe request with the same notification_id
+    result = is_duplicate_unsubscribe_request(notification_id)
+    assert result is True
+
+
+def test_is_duplicate_unsubscribe_request_for_batched_unprocessed_request(sample_service):
+    notification_id = create_unsubscribe_request_and_return_the_notification_id(
+        sample_service, is_batched=True, processed_by_service=False
+    )
+    # Simulate a duplicate unsubscribe request with the same notification_id
+    result = is_duplicate_unsubscribe_request(notification_id)
+    assert result is True
```

Comment:
Think this would be easier to follow if you rewrote it as one test with 
```python
@pytest.mark.parametrize("is_batched, proccessed_by_service, expected_duplicate", (…))
```

Reply:
Agreed.